fbw-a380x/src/systems/instruments/src/MFD/FMC/FmcAircraftInterface.tsx

```diff
@@ -803,15 +806,57 @@ export class FmcAircraftInterface {
       this.managedSpeedTargetIsMach = isMach;
     }
 
-    // Overspeed protection
-    const Vtap = Math.min(this.managedSpeedTarget ?? Vmo, SimVar.GetSimVarValue('L:A32NX_SPEEDS_VMAX', 'number'));
+    let Vtap = 0;
+    // VLS protection
+    if (this.managedSpeedTarget) {
+      const vls = this.speedVls.get();
+      if (this.managedSpeedTarget < vls) {
+        Vtap = vls;
+      }
+    }
 
+    if (!Vtap) {
+      // Overspeed protection
+      const vMax = SimVar.GetSimVarValue('L:A32NX_SPEEDS_VMAX', 'number');
+      Vtap = Math.min(this.managedSpeedTarget ?? Vmo - 5, vMax - 5);
+    }
     SimVar.SetSimVarValue('L:A32NX_SPEEDS_MANAGED_PFD', 'knots', vPfd);
     SimVar.SetSimVarValue('L:A32NX_SPEEDS_MANAGED_ATHR', 'knots', Vtap);
 
-    if (this.isAirspeedManaged()) {
+    const ismanaged = this.isAirspeedManaged();
+
+    if (ismanaged) {
       Coherent.call('AP_SPD_VAR_SET', 0, Vtap).catch(console.error);
     }
+
+    //short term managed speed
+    let shortTermManagedSpeed = 0;
+    const phase = this.fmgc.getFlightPhase();
+    if (phase != FmgcFlightPhase.Preflight) {
+      if (this.managedSpeedTarget) {
+        if (ismanaged) {
+          const verticalMode = SimVar.GetSimVarValue('L:A32NX_FMA_VERTICAL_MODE', 'number');
+          const shortTermActiveInmanaged =
+            !takeoffGoAround && this.fmgc.getFlightPhase() != FmgcFlightPhase.Cruise && verticalMode != 23; // DES
+          if (shortTermActiveInmanaged && this.isSpeedDifferenceGreaterThan2Kt(vPfd, Vtap)) {
+            shortTermManagedSpeed = Vtap;
+          }
+        } else {
+          const selectedSpeed = SimVar.GetSimVarValue('L:A32NX_AUTOPILOT_SPEED_SELECTED', 'number');
+          if (selectedSpeed) {
+            const speedTarget = phase == FmgcFlightPhase.Approach ? Vtap : vPfd; // FIX me Should use ECON during hold & deceleration segments
+            if (this.isSpeedDifferenceGreaterThan2Kt(selectedSpeed, speedTarget)) {
+              shortTermManagedSpeed = speedTarget;
+            }
+          }
+        }
+      }
+    }
+    SimVar.SetSimVarValue('L:A32NX_SPEEDS_MANAGED_SHORT_TERM_PFD', 'knots', shortTermManagedSpeed);
```

Comment:
Use a subject and rounding to avoid writing vars unnecessarily. See https://github.com/flybywiresim/aircraft/pull/8965/files for inspiration.